File: Year-2025 Problem List/February 2025/3151. Special Array I/3151. Special Array I.cpp
/*
# Problem          : 3151. Special Array I
# Time Complexity  : O(N)
# C++ Solution
*/
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        int n = (int)nums.size();
        for (int i = 0; i < n - 1; i++) {
            if (nums[i] % 2 == nums[i + 1] % 2) {
                return false;
            }
        }
        return true;
    }
};

File: Year-2025 Problem List/February 2025/3151. Special Array I/3151. Special Array I.py
# Problem          : 3151. Special Array I
# Time Complexity  : O(n)
# Python Solution

class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        for i in range(len(nums) - 1):
            if nums[i] % 2 == nums[i + 1] % 2:
                return False
        return True
